java/maven.indexer/src/org/netbeans/modules/maven/indexer/ClassDependencyIndexCreator.java

```diff
@@ -256,34 +256,36 @@ private static Set<String> parseField(String refereeCRC, String field, String re
         "org/omg/CosNaming", "org/omg/Dynamic", "org/omg/DynamicAny", "org/omg/IOP", "org/omg/Messaging",
         "org/omg/PortableInterceptor", "org/omg/PortableServer", "org/omg/SendingContext", "org/omg/stub",
         "org/w3c/dom", "org/xml/sax"
-    });
+    };
 
     /**
      * @param referrer a referring class, as {@code pkg/Outer$Inner}
-     * @param data its bytecode
+     * @param classData its bytecode
      * @param depsMap map from referring outer classes (as {@code pkg/Outer}) to referred-to classes (as {@code pkg/Outer$Inner})
      * @param siblings other referring classes in the same artifact (including this one), as {@code pkg/Outer$Inner}
      */
-    private static void addDependenciesToMap(String referrer, InputStream data, Map<String, Set<String>> depsMap, Set<String> siblings) throws IOException {
+    private static void addDependenciesToMap(String referrer, InputStream classData, Map<String, Set<String>> depsMap, Set<String> siblings) throws IOException {
+
         int shell = referrer.indexOf('$', referrer.lastIndexOf('/') + 1);
         String referrerTopLevel = shell == -1 ? referrer : referrer.substring(0, shell);
-        for (String referee : dependencies(data)) {
-            if (referrer.equals(referee)) {
-                continue;
-            }
-            if (siblings.contains(referee)) {
-                continue; // in same JAR, not interesting
-            }
-            if (JDK_CLASS_TEST.test(referee)) {
-                continue;
-            }
-            Set<String> referees = depsMap.get(referrerTopLevel);
-            if (referees == null) {
-                referees = new HashSet<>();
-                depsMap.put(referrerTopLevel, referees);
-            }
-            referees.add(referee);
+
+        Set<String> tmp = depsMap.get(referrerTopLevel);
+        if (tmp == null) {
+            tmp = new HashSet<>();
+            depsMap.put(referrerTopLevel, tmp);
         }
+        Set<String> referees = tmp;
+
+        dependenciesOf(classData)
+            .filter((referee) -> !referrer.equals(referee))
+            .filter((referee) -> !siblings.contains(referee)) // in same JAR, not interesting
+            .filter((referee) -> {
+                for (int i = 0; i < JDK_CLASS_TEST.length; i++)
+                    if (referee.startsWith(JDK_CLASS_TEST[i]))
+                        return false;
```

Comment:
i am going to add the braces the next time I look at this class if I remember. I usually try to keep Stream expressions as compact as possible, in this case it would be even clearer if the filter lambda is extracted to a method - with all the braces there.